### PR TITLE
declare customized features from config

### PR DESCRIPTION
**Pull request by pfliu-nlp**
This PR aims to make customize features from datalab be declared from a [config file](https://github.com/neulab/ExplainaBoard/blob/add_customized_features_from_config/explainaboard/resources/dataset_custom_features.json) and apply this functionality to open-domain QA task.

`dataset_custom_features.json`
```json
{
  "sst2": {
    "label": {
      "dtype": "string",
      "description": "the true label",
      "num_buckets": 2
    }
  },
  "natural_questions_comp_gen": {
    "question_types": {
      "dtype": "string",
      "description": "the question type",
      "num_buckets": 5
    }
  }
}

```

### CODE REVIEW

File: explainaboard/loaders/file_loader.py [hunk 1]
```diff
@@ -526,6 +527,15 @@ def load_raw(
         self, data: str | DatalabLoaderOption, source: Source
     ) -> FileLoaderReturn:
         config = narrow(DatalabLoaderOption, data)
+
+        # load customized features from global config files
+        customized_features_from_config = get_custmomized_features()
```
Comment (neubig) [inline explainaboard/loaders/file_loader.py:532]: typo: should be "customized"

File: explainaboard/loaders/file_loader.py [hunk 2]
```diff
@@ -526,6 +527,15 @@ def load_raw(
...
+        if config.dataset in customized_features_from_config.keys():
```
Comment (neubig) [inline explainaboard/loaders/file_loader.py:533]:
```suggestion
        if config.dataset in customized_features_from_config:
```

File: explainaboard/loaders/file_loader.py [hunk 3]
```diff
@@ -547,6 +557,14 @@ def load_raw(
 
         # Infer metadata from the dataset
         metadata = FileLoaderMetadata()
+        # load customized features from global config files
+        if config.dataset in customized_features_from_config.keys():
```
Comment (neubig) [inline explainaboard/loaders/file_loader.py:561]:
```suggestion
        if config.dataset in customized_features_from_config:
```

File: explainaboard/resources/dataset_custom_features.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (neubig) [inline explainaboard/resources/dataset_custom_features.json:1]: Could you please add the features for `sumeval2022` as well?

**Review by neubig — COMMENTED** [review]
Just a few suggestions.

File: explainaboard/loaders/file_loader.py (hunk 3 above)
Comment (pfliu-nlp) [reply explainaboard/loaders/file_loader.py:561]: I think `if config.dataset in customized_features_from_config.keys()` is correct here.

File: explainaboard/loaders/file_loader.py (hunk 2 above)
Comment (pfliu-nlp) [reply explainaboard/loaders/file_loader.py:533]: I think `if config.dataset in customized_features_from_config.keys()` is correct here.

File: explainaboard/loaders/file_loader.py (hunk 3 above)
Comment (neubig) [reply explainaboard/loaders/file_loader.py:561]: Sorry, I wasn't clear enough on this. I think they're both functionally identical, but:
1. writing `.keys()` is more verbose, and
2. I believe `if x in y.keys()` is `O(len(y))`, where `if x in y` is `O(1)`, as the former returns a view of the keys as a list: https://www.w3schools.com/python/ref_dictionary_keys.asp

So I'd suggest not writing `.keys()`

Comment (neubig) [reply explainaboard/loaders/file_loader.py:561]: Actually I was wondering about "2." so I researched a bit more, and it seems that starting in Python 3.x both are `O(1)`: https://stackoverflow.com/a/17539423

But still, not writing `.keys()` is less verbose, so I'd recommend it either way.

Comment (pfliu-nlp) [reply explainaboard/loaders/file_loader.py:561]: oh, I see. Thanks for the patient explanation!

**Review by neubig — APPROVED** [review]
LGTM.

**Pull request merged**